src/cc_catalog_airflow/dags/provider_api_scripts/brooklyn_museum.py

```diff
@@ -0,0 +1,190 @@
+import os
+import re
+import logging
+from common.requester import DelayedRequester
+from common.storage.image import ImageStore
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)s - %(levelname)s:  %(message)s',
+    level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+LIMIT = 35
+DELAY = 5.0
+RETRIES = 3
+PROVIDER = "brooklynmuseum"
+ENDPOINT = "https://www.brooklynmuseum.org/api/v2/object/"
+API_KEY = os.getenv("BROOKLYN_MUSEUM_API_KEY", "nokeyprovided")
+
+delay_request = DelayedRequester(delay=DELAY)
+image_store = ImageStore(provider=PROVIDER)
+
+HEADERS = {
+    "api_key": API_KEY
+}
+
+DEFAULT_QUERY_PARAM = {
+    "has_images": 1,
+    "rights_type_permissive": 1,
+    "limit": LIMIT,
+    "offset": 0
+}
+
+
+def main():
+    logger.info("Begin: Brooklyn museum provider script")
+    offset = 0
+    condition = True
+    while condition:
+        query_param = _get_query_param(offset)
+        objects_batch = _get_object_json(query_param=query_param)
+        logger.debug(len(objects_batch))
+        if len(objects_batch) > 0:
+            image_count = _process_objects_batch(objects_batch)
+            logger.debug(f"Images till now {image_count}")
+            offset += LIMIT
+        else:
+            condition = False
+    total_images = image_store.commit()
+    logger.info(f"Total images recieved {total_images}")
+
+
+def _get_query_param(offset=0,
+                     default_query_param=DEFAULT_QUERY_PARAM):
+    query_param = default_query_param.copy()
+    query_param.update(
+        offset=offset
+    )
+    return query_param
+
+
+def _get_object_json(headers=HEADERS,
+                     endpoint=ENDPOINT,
+                     retries=RETRIES,
+                     query_param=None):
+    response_json = None
+    for tries in range(retries):
+        try:
+            response = delay_request.get(
+                endpoint,
+                query_param,
+                headers=headers
+            )
+            data = response.json()
+            if data and data.get("message", "").lower() == "success.":
+                response_json = data.get("data", "")
+                break
+        except Exception as e:
+            logger.warning(f"response not captured due to {e}")
+    return response_json
+
+
+def _process_objects_batch(objects_batch):
+    image_count = None
+    for object_ in objects_batch:
+        rights_info = object_.get("rights_type")
+        license_url = _get_license_url(rights_info)
+        logger.debug(license_url)
+        if license_url is not None:
+            id_ = object_.get("id", "")
+            complete_object_data = _get_object_json(
+                endpoint=ENDPOINT+str(id_))
+            if complete_object_data is None:
+                continue
+            image_count = _handle_object_data(data=complete_object_data,
+                                              license_url=license_url)
```

Comment:
This style is more modifiable later (with smaller diffs):
```suggestion
            image_count = _handle_object_data(
                data=complete_object_data,
                license_url=license_url
            )
```